src/blocks/Block.as

```diff
@@ -550,9 +464,18 @@ public class Block extends Sprite {
 				}
 			}
 		}
-		if (nextBlock != null) dup.addChild(dup.nextBlock = nextBlock.duplicate(forClone, forStage));
-		if (subStack1 != null) dup.addChild(dup.subStack1 = subStack1.duplicate(forClone, forStage));
-		if (subStack2 != null) dup.addChild(dup.subStack2 = subStack2.duplicate(forClone, forStage));
+		if (nextBlock != null) {
+			dup.nextBlock = nextBlock.duplicate(forClone, forStage);
+			dup.nextBlock.prevBlock = dup;
+		}
+		if (subStack1 != null) {
+			dup.subStack1 = subStack1.duplicate(forClone, forStage);
+			dup.subStack1.prevBlock = dup;
```

Comment:
I did this to have non-DOM references between the Blocks. When the user switches to view the duplicate sprites blocks they will get added to the ScriptPane within BlockStack containers. At that point they will be added to the DOM.